.github/workflows/test.yml

```diff
@@ -6,8 +6,24 @@ name: Run Tests
 on:
   push:
     branches: [ dev ]
+    inputs:
+      testEnv:
+        description: 'Environment in which tests should be run. Currently works on dev, alpha and staging'
+        required: true
+        default: 'dev'
   pull_request:
     branches: [ '**' ]
+    inputs:
+      testEnv:
+        description: 'Environment in which tests should be run. Currently works on dev, alpha and staging'
+        required: true
+        default: 'local'
+  workflow_dispatch:
+    inputs:
+      testEnv:
+        description: 'Environment in which tests should be run. Currently works on dev, alpha and staging'
+        required: true
+        default: 'dev'
```

Comment:
can the `inputs` specification be moved up a level, so it's not repeated for each trigger? ( I haven't tried, don't know if it's supported )

Reply:
I think only dispatched events have inputs, I don't think they exist for PR or push.